web/packages/teleport/src/Users/UserAddEdit/TraitsEditor.tsx

```diff
@@ -0,0 +1,271 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, Dispatch, SetStateAction } from 'react';
+import { ButtonBorder, Box, Flex, Text, ButtonIcon } from 'design';
+import { Add, Trash } from 'design/Icon';
+import { FieldSelectCreatable } from 'shared/components/FieldSelect';
+import { Option } from 'shared/components/Select';
+import { requiredField, requiredAll } from 'shared/components/Validation/rules';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+
+import { AllUserTraits } from 'teleport/services/user';
+
+/**
+ * traitsPreset is a list of system defined traits in Teleport.
+ * The list is used to populate traits key option.
+ */
+const traitsPreset = [
+  'aws_role_arns',
+  'azure_identities',
+  'db_names',
+  'db_roles',
+  'db_users',
+  'gcp_service_accounts',
+  'host_user_gid',
+  'host_user_uid',
+  'kubernetes_groups',
+  'kubernetes_users',
+  'logins',
+  'windows_logins',
+];
+
+/**
+ * TraitsEditor supports add, edit or remove traits functionality.
+ * @param allTraits all traits pre-configured for user.
+ * @param attempt attempt is Attempt status.
+ * @param configuredTraits holds traits configured for user in current editor.
+ * @param setConfiguredTraits sets user traits in current editor.
+ */
+export function TraitsEditor({
+  allTraits,
+  attempt,
+  configuredTraits,
+  setConfiguredTraits,
+}: TraitEditorProps) {
+  useEffect(() => {
+    const newTraits = traitsToTraitsOption(allTraits);
+    setConfiguredTraits(newTraits);
+  }, [allTraits]);
+
+  function handleInputChange(i: InputOption) {
+    let newTraits = [...configuredTraits];
```

Comment:
I have a mixed feeling on using `const` in this case. Cause, `const` semantically means the value wont change. But here we change `newTraits` value and that is only possible because its an array. So using `let` conveys that its value changes below.

Reply:
i can see how that can be confusing 😅 , but i think this [article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) may help you out

tldr: 

> The const declaration creates an immutable reference to a value. It does not mean the value it holds is immutable — just that the variable identifier cannot be reassigned.